external-providers/yq-external-provider/pkg/generic/service_client.go

```diff
@@ -0,0 +1,317 @@
+package generic
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/konveyor/analyzer-lsp/jsonrpc2"
+	"github.com/konveyor/analyzer-lsp/lsp/protocol"
+	"github.com/konveyor/analyzer-lsp/provider"
+	"go.lsp.dev/uri"
+	"golang.org/x/mod/semver"
+	"gopkg.in/yaml.v2"
+)
+
+type genericServiceClient struct {
+	rpc        *jsonrpc2.Conn
+	cancelFunc context.CancelFunc
+	log        logr.Logger
+	cmd        *exec.Cmd
+
+	config       provider.InitConfig
+	capabilities protocol.ServerCapabilities
+}
+
+type Release struct {
+	Name string `json:"name"`
+}
+
+const APIVERSION = "apiVersion"
+const KIND = "kind"
+const GITHUB_K8S_API_URL = "https://api.github.com/repos/kubernetes/kubernetes/releases"
+
+var default_k8s_version = "v1.28.4"
+
+var _ provider.ServiceClient = &genericServiceClient{}
+
+func (p *genericServiceClient) Stop() {
+	p.cancelFunc()
+	p.cmd.Wait()
+}
+
+func (p *genericServiceClient) Evaluate(ctx context.Context, cap string, conditionInfo []byte) (provider.ProviderEvaluateResponse, error) {
+	var cond genericCondition
```

Comment:
Maybe I'm not quite understanding, but does this support both "referenced" and "k8sResourceMatched" like stated in the `Capabilities` function? Looks like only the latter is supported here, which is fine. May need to update the `Capabilities` function

Reply:
I think just need to keep k8sResourceMatched Capabilities for yq provider